LuzyceApi/Controllers/LoginController.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using LuzyceApi.Data;
+using LuzyceApi.Dtos.User;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace LuzyceApi.Controllers
+{
+    [Route("api/login")]
+    [ApiController]
+    public class LoginController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IConfiguration _config;
+
+        public LoginController(ApplicationDbContext context, IConfiguration config)
+        {
+            _context = context;
+            _config = config;
+        }
+
+        private string GenerateJSONWebToken(int userId)
+        {
+            var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_config["Jwt:SigningKey"] ?? ""));
+            var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+            var token = new JwtSecurityToken(
+                issuer: _config["Jwt:Issuer"],
+                expires: DateTime.Now.AddHours(1),
+                signingCredentials: credentials
+            );
+
+            return new JwtSecurityTokenHandler().WriteToken(token);
+        }
+
+        [HttpPost]
+        public IActionResult Login([FromBody] LoginDto dto)
+        {
+            var user = _context.Users.FirstOrDefault(x => x.Login == dto.Login && x.Password == dto.Password);
```

Comment:
hasło nie powinno być zapisane w bazie w plaintexcie.
tutaj powinieneś pobrać usera z loginem dto.Login i sprawdzić czy hasło się zgadza.

Installing BCrypt.Net-Next via NuGet
.NET CLI: `dotnet add package BCrypt.Net-Next`

Visual Studio Package Manager Console: Install-Package BCrypt.Net-Next


Hashing a password in .NET 6.0
The following line of code hashes the password "Pa$$w0rd" using bcrypt and stores the result in the passwordHash string variable.

`string passwordHash = BCrypt.Net.BCrypt.HashPassword("Pa$$w0rd");`

Verify a password against a hash in .NET 6.0
The following line of code verifies the password "Pa$$w0rd" using bcrypt against the hash stored in the passwordHash variable.

`bool verified = BCrypt.Net.BCrypt.Verify("Pa$$w0rd", passwordHash);`